wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/WikibaseDataEditor.java

```diff
@@ -530,13 +533,13 @@ public ItemDocument editItemDocument(ItemDocument itemDocument,
 	 * @throws MediaWikiApiErrorException
 	 *             if MediaWiki API returned an error response
 	 */
-	public void editEntityDocument(
+	public EditingResult editEntityDocument(
 			EntityUpdate update, boolean clear, String summary, List<String> tags)
 			throws IOException, MediaWikiApiErrorException {
 		long revisionId = update.getBaseRevisionId();
 		if (!clear) {
 			if (update.isEmpty())
-				return;
+				return new EditingResult(0L);
```

Comment:
Shouldn't this be the current revision ID rather than zero?

Reply:
I considered that but did not go for it because the update might not have a base revision id in the first place. I want to be able to tell from the response if an edit was made or not.

I guess we could forward the base revision id and add another field on `EditingResult` to record the number of edits (= API actions) used to carry out the update. At the moment this would be either 0 or 1, but there could be scenarios where we cannot or do not want to do an update atomically (for instance to get more meaningful edit summaries).

Reply:
Ah, okay, I thought it's an accidental omission. Now it's clear there's reasoning behind it. You might want to consider optionals, so that people can provide their own fallback in `orElse()`.